code/modules/mob/living/simple_animal/bot/hygienebot.dm

```diff
@@ -0,0 +1,356 @@
+	if(washing)
+		wash_atom(AM)
+
+/mob/living/simple_animal/bot/hygienebot/update_icon()
+	cut_overlays()
+
+	if(on)
+		var/mutable_appearance/fire_overlay = mutable_appearance(icon,"flame")
+		add_overlay(fire_overlay)
+		icon_state = "drone-on"
+	else
+		icon_state = "drone"
+	if(washing)
+		var/mutable_appearance/water_overlay = mutable_appearance(icon, emagged ? "dronefire" : "dronewater")
+		add_overlay(water_overlay)
+
+/mob/living/simple_animal/bot/hygienebot/turn_off()
+	..()
+	mode = BOT_IDLE
+
+/mob/living/simple_animal/bot/hygienebot/bot_reset()
+	..()
+	target = null
+	oldtarget_name = null
+	walk_to(src,0)
+	last_found = world.time
+
+
+/mob/living/simple_animal/bot/hygienebot/Crossed(atom/movable/AM)
+	. = ..()
+	if(washing)
+		wash_atom(AM)
+
+/mob/living/simple_animal/bot/hygienebot/Moved()
+	. = ..()
+	if(washing && isturf(loc) && !emagged)
+		var/turf/open/OT = loc
+		OT.MakeSlippery(TURF_WET_WATER, min_wet_time = 10 SECONDS, wet_time_to_add = 5 SECONDS)
+
+/mob/living/simple_animal/bot/hygienebot/handle_automated_action()
+	if(!..())
+		return
+
+	if(washing)
+		wash_atom(loc)
+		for(var/AM in loc)
+			wash_atom(AM)
+		if(isopenturf(loc) && !emagged)
+			var/turf/open/tile = loc
+			tile.MakeSlippery(TURF_WET_WATER, min_wet_time = 10 SECONDS, wet_time_to_add = 5 SECONDS)
+
+	switch(mode)
+		if(BOT_IDLE)		// idle
+			walk_to(src,0)
+			look_for_lowhygiene()	// see if any disgusting fucks are in range
+			if(!mode && auto_patrol)	// still idle, and set to patrol
+				mode = BOT_START_PATROL	// switch to patrol mode
+
+		if(BOT_HUNT)		// hunting for stinkman
+			// if can't reach stinkman for long enough, don't give up, try harder.
+			if(emagged) //lol fuck em up
+				currentspeed = 8
+				start_washing()
+				mad = TRUE
+			else
+				switch(frustration)
+					if(0 to 4)
+						currentspeed = 5
+						stop_washing()
+						mad = FALSE
+					if(4 to INFINITY)
+						currentspeed = 2.5
+						start_washing()
+						mad = TRUE
+			if(target)
+				if(target.loc == loc && isturf(target.loc)) //LADIES AND GENTLEMAN WE GOTEM PREPARE TO DUMP
+					start_washing()
+					if(mad)
+						speak("Well about fucking time you degenerate", "Fucking finally", "Thank god, you finally stopped")
+						playsound(loc, 'sound/effects/hygienebot_angry.ogg', 60, 1)
+						mad = FALSE
+					mode = BOT_SHOWERSTANCE
+				else
+					var/turf/olddist = get_dist(src, target)
+					walk_to(src, target,0, currentspeed)
+					if(mad && prob(60))
+						playsound(loc, 'sound/effects/hygienebot_angry.ogg', 60, 1)
+						speak(pick("Get back here you foul smelling fucker.", "If you don't get back here right now I'm going to give you a fucking vasectomy.", "STOP RUNNING OR I WILL CUT YOUR ARTERIES!", "Just fucking let me clean you you arsehole!", "STOP. RUNNING.", "Either you stop running or I will fucking drag you out of an airlock.", "I just want to fucking clean you you troglodyte.", "If you don't come back here I'll put a green cloud around you cunt."))
```

Comment:
isn't the point of a showerbot to remove green clouds from the cunt

Reply:
it's a meta joke about how i made green clouds because noone would wash blood of off their guys